vnext/Shared/WinRTWebSocketResource.cpp

```diff
@@ -123,13 +117,14 @@ WinRTWebSocketResource::~WinRTWebSocketResource() noexcept /*override*/
 
 #pragma region Private members
 
-IAsyncAction WinRTWebSocketResource::PerformConnect() noexcept {
+IAsyncAction WinRTWebSocketResource::PerformConnect(Uri &&uri) noexcept {
   auto self = shared_from_this();
+  auto coUri = std::move(uri);
```

Comment:
>= std::move(uri); [](http://example.com/codeflow?start=13&length=17)

I wonder what the result of this operation is. Are you sure that the Uri constructor is called and we not keep just a reference?
I would rather be explicit. E.g.
Uri coUri{std::move(uri)};

Reply:
I can triple-check, but not keeping this reference made the `uri` inaccessible by the coroutine when it resumed.
In short, it seemed like the value was not being passed to the resuming thread.

Reply:
Verified. Not keeping a new reference crashes tests.